File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost1 = sorted(lost)
    reserve1 = sorted(reserve)
    lost2 = set(lost1)-set(reserve1)
    reserve2 = set(reserve)-set(lost1)
    for i in reserve2:
        if i-1 in lost2:
            lost2.remove(i-1)
            continue
        if i+1 in lost2:
            lost2.remove(i+1)
    
    return n - len(lost2)

File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    lost.sort()
    reserve.sort()
    let realLost = lost.filter(element => !reserve.includes(element))
    let realReserve = reserve.filter(element => !lost.includes(element))
    const sucessReserve = []
    realLost.forEach(item =>{
        if (realReserve.includes(item-1)) {
            realReserve = realReserve.filter(element => element !== item -1)
            sucessReserve.push(item-1)
        }
        else if(realReserve.includes(item+1)){ 
            realReserve = realReserve.filter(element => element !== item + 1)
            sucessReserve.push(item+1)
        }
    })
    
    const set = new Set(sucessReserve)
    return Math.min((n - realLost.length + set.size),n)
                
}


// 1. reverse / lost에서 중복되면 제거한다.
